src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -858,6 +903,18 @@ private static double ComputeWeightedDistance(ForEachStatementSyntax left, ForEa
             return AdjustForLocalsInBlock(distance, left.Statement, right.Statement, localsWeight: 0.6);
         }
 
+        private static double ComputeWeightedDistance(ForEachComponentStatementSyntax left, ForEachComponentStatementSyntax right)
```

Comment:
Should be merged with distance calculation for regular ForEachStatement and handle transitions between these two forms.